plugins/cpp/test/src/cppparsertest.cpp

```diff
@@ -526,10 +538,15 @@ TEST_F(CppParserTest, Enum)
           break;
       }
     }
+  });
+}
 
-    model::CppFunction fieldFunction = _db->query_value<model::CppFunction>(
+TEST_F(CppParserTest, Fields)
+{
+  _transaction([&, this] {
+    model::CppVariable fieldFunction = _db->query_value<model::CppVariable>(
       QCppFunction::name == "fieldFunction");
-    astNodes = _db->query<model::CppAstNode>(
+    RCppAstNode astNodes = _db->query<model::CppAstNode>(
```

Comment:
This is the case where I was uncertain about my fix. This `fieldFunction` is a defined like this:

```cpp
struct Derived : public MyClass
{
  MyClass fieldVariable;
  MyClass (*fieldFunction)(const MyClass&);
  
  // etc...
};
```

So it is a class member field, as a function pointer. It gets parsed into the `CppVariable` table, not the `CppFunction`, but that is correct that way, right @bruntib ? It is a variable, not a function semantically.

Reply:
I checked the above code, the function pointer is considered a `fieldDecl`.
![image](https://user-images.githubusercontent.com/26124787/112617176-e9f44a00-8e24-11eb-9a72-3ff757943531.png)